packages/pageflow-react/src/pages/createPageType.jsx

```diff
@@ -47,6 +48,11 @@ export default function({Component,
         id: pageId(pageElement),
         position: options.position
       }));
+      if(!!configuration.hide_logo) {
+        jQuery('.header_logo').stop( true, true ).fadeOut(1000);
+      } else {
+        jQuery('.header_logo').stop( true, true ).fadeIn(1000);
+      };
```

Comment:
Doing it here, will only make it work for React based pages - not the legacy ones. 

What we are trying to achieve is very similar to toggling between the inverted/normal logo variant based on the inverted setting of the page. You already found the `updateCommonPageCssClasses` function which does this for the background image logo. The watermark logo is part of the header, which has its own jQuery widget `packages/pageflow/src/frontend/jquery_ui_widgets/header.js` to [handle such logic](https://github.com/codevise/pageflow/blob/9872a63d8e7b527b4cfaaa6eaeb8019fc090b3d3/packages/pageflow/src/frontend/jquery_ui_widgets/header.js#L37). The corresponding CSS for the header lives [here](https://github.com/codevise/pageflow/blob/9872a63d8e7b527b4cfaaa6eaeb8019fc090b3d3/app/assets/stylesheets/pageflow/themes/default/logo/variant/watermark.scss#L100), where we can also place a new CSS class that hides the logo instead of using jQuery animations.